File: src/main/java/nars/entity/BudgetTriple.java
package nars.entity;

public interface BudgetTriple {
    float getPriority();

    void setPriority(float v);

    void incPriority(float v);

    void decPriority(float v);

    float getDurability();

    void setDurability(float v);

    void incDurability(float v);

    void decDurability(float v);

    float getQuality();

    void setQuality(float v);

    void merge(BudgetTriple that);

//    float summary();
//
//    boolean aboveThreshold();
}

File: src/main/java/nars/entity/BudgetTriple.kt
package nars.entity

/**
 *
 */
interface BudgetTriple:Iterable<Float> {
    /**
     *
     */
    var priority: Float

    /**
     *
     */
    fun incPriority(v: Float)

    /**
     *
     */
    fun decPriority(v: Float)

    /**
     *
     */
    var durability: Float

    /**
     *
     */
    fun incDurability(v: Float)

    /**
     *
     */
    fun decDurability(v: Float)

    /**
     *
     */
    var quality: Float

    /**
     *
     */
    fun merge(that: BudgetTriple ) :BudgetTriple

}
